gtsfm/frontend/correspondence_generator/colmap_correspondence_generator.py

```diff
@@ -49,24 +49,42 @@ def __init__(self, database_path: str) -> None:
             self._pycolmap_db.num_verified_image_pairs,
         )
 
-    def _read_keypoints(self, pycolmap_image_id: int) -> Keypoints:
+    def _read_keypoints(self, image: Image) -> Keypoints:
         """Read keypoints from pycolmap.Image object."""
-        if pycolmap_image_id not in self._keypoints_dict:
+        pycolmap_image = self._pycolmap_db.read_image_with_name(image.file_name)
+        image_id = pycolmap_image.image_id
+        if image_id not in self._keypoints_dict:
             return Keypoints(coordinates=np.array([], dtype=np.float32))
-
-        return Keypoints(coordinates=self._keypoints_dict[pycolmap_image_id][:, :2], scales=None, responses=None)
-
-    def _read_image_ids_and_keypoints(self, images: List[Image]) -> Tuple[List[int], List[Keypoints]]:
+        coordinates = self._keypoints_dict[image_id][:, :2]
+        camera = self._pycolmap_db.read_camera(pycolmap_image.camera_id)
+        # Colmap extracts features in the downscaled image 
+        # but scales keypoints back to the original dimensions before storing in the database.
+        if image.width != camera.width or image.height != camera.height:
+            scale = np.array([image.width / camera.width, image.height / camera.height])
+            return Keypoints(
+                coordinates=coordinates * scale, scales=None, responses=None
+            )
+        return Keypoints(coordinates=coordinates, scales=None, responses=None)
+
+    def _read_image_ids_and_keypoints(
```

Comment:
are you using the same formatting settings? please refer to https://github.com/borglab/gtsfm/blob/master/CONTRIBUTING.md#python-style